crates/ir-generated/src/emit/emitter.rs

```diff
@@ -0,0 +1,87 @@
+use std::mem;
+
+use rokugo_ir::{
+    container::IrContainer,
+    op_code::IrOpCode,
+    register::{chill::RegisterChill, natural::RegisterNat32, Register, RegisterId},
+};
+
+use crate::errors::register::RegisterAllocationError;
+
+use super::register_allocator::{RegisterAllocator, RegisterDropGuard};
+
+pub struct IrEmitter {
+    data: Vec<u8>,
+    register_allocator: RegisterAllocator,
+}
+
+impl IrEmitter {
+    pub fn new() -> Self {
+        IrEmitter {
+            data: Vec::new(),
+            register_allocator: RegisterAllocator::new(),
+        }
+    }
+}
+
+/// # Local Memory
+impl IrEmitter {
+    /// Allocates a virtual register, or prepare a native register to store a new 32-bit natural value.
+    pub fn alloc_register_nat32(
+        &mut self,
+        chill: RegisterChill,
+    ) -> Result<RegisterDropGuard<RegisterNat32>, RegisterAllocationError> {
+        let id = self.register_allocator.next_nat32()?;
+        self.emit(IrOpCode::AllocRegisterNat32);
+        self.emit_register_id(id);
+        self.data.extend_from_slice(&chill.to_le_bytes());
+
+        // SAFETY: This is safe, because `RegisterNat32` is a wrapper around `RegisterId`.
+        Ok(RegisterDropGuard::new(unsafe { mem::transmute(id) }))
+    }
+
+    /// Drops a virtual register, what preverts it from being chilled.
+    /// # Remarks
+    /// This instruction is not dropping any memory like pointer etc. It is only a hint to the JIT compiler.
```

Comment:
Please include a blank line before and after headers in code comments. It doesn't change the rustdoc output, but helps readability when you're browsing the code.
```suggestion
    /// Drops a virtual register, what preverts it from being chilled.
    ///
    /// # Remarks
    ///
    /// This instruction is not dropping any memory like pointer etc. It is only a hint to the JIT compiler.
```